### PR TITLE
Flip trailing classpathOrder params to true

### PR DESCRIPTION
**Pull request by dwijnand**
Refs #1466

### CODE REVIEW

**Comment by alexarchambault** [comment]
I'm not sure this should be changed for the `resolve` command itself. That makes its output confusing IMO. As it's meant to be read by humans in particular.

But it makes sense for the other commands (`fetch`, `launch`, `bootstrap`).

I was more thinking about changing `ResolveOptions.classpathOrder` to an option (same thing in `ResolveParams`), and have each command ([`resolve`](https://github.com/coursier/coursier/blob/cd65765a1806f37562b07e69972b50feda267fc2/modules/cli/src/main/scala/coursier/cli/resolve/Output.scala#L92-L95), [`fetch`](https://github.com/coursier/coursier/blob/cd65765a1806f37562b07e69972b50feda267fc2/modules/cli/src/main/scala/coursier/cli/fetch/Fetch.scala#L48), [`launch`](https://github.com/coursier/coursier/blob/cd65765a1806f37562b07e69972b50feda267fc2/modules/cli/src/main/scala/coursier/cli/launch/Launch.scala#L269), [`bootstrap`](https://github.com/coursier/coursier/blob/cd65765a1806f37562b07e69972b50feda267fc2/modules/cli/src/main/scala/coursier/cli/bootstrap/Bootstrap.scala#L227)) call `getOrElse` with its sensible default.

**Comment by dwijnand** [comment]
Hmm.  Don't you think it's more confusing for users that these commands would behave differently?  I kind of feel that I (and anyone like me with no details on implementation details) would reasonably expect that the order returned by something like `resolve` to be the order of deps used by launch.

What if we change the help of resolve and mention something about how `--classpathOrder false` might be more pleasant to read?

**Comment by alexarchambault** [comment]
I agree that that would be more coherent, yet the output of `resolve` is just unreadable to me with classpath order.

Contrast
```bash
$ cs resolve ammonite --classpath-order=false
com.github.javaparser:javaparser-core:3.2.5:default
com.github.scopt:scopt_2.12:3.7.1:default
com.lihaoyi:ammonite-interp-api_2.12.10:2.0.4:default
com.lihaoyi:ammonite-interp_2.12.10:2.0.4:default
com.lihaoyi:ammonite-ops_2.12:2.0.4:default
com.lihaoyi:ammonite-repl-api_2.12.10:2.0.4:default
com.lihaoyi:ammonite-repl_2.12.10:2.0.4:default
com.lihaoyi:ammonite-runtime_2.12.10:2.0.4:default
com.lihaoyi:ammonite-terminal_2.12:2.0.4:default
com.lihaoyi:ammonite-util_2.12:2.0.4:default
com.lihaoyi:ammonite_2.12.10:2.0.4:default
com.lihaoyi:fansi_2.12:0.2.8:default
com.lihaoyi:fastparse_2.12:2.2.3:default
com.lihaoyi:geny_2.12:0.5.0:default
com.lihaoyi:os-lib_2.12:0.6.3:default
com.lihaoyi:pprint_2.12:0.5.8:default
com.lihaoyi:requests_2.12:0.5.0:default
com.lihaoyi:scalaparse_2.12:2.2.3:default
com.lihaoyi:sourcecode_2.12:0.2.0:default
com.lihaoyi:ujson_2.12:0.9.8:default
com.lihaoyi:upack_2.12:0.9.8:default
com.lihaoyi:upickle-core_2.12:0.9.8:default
com.lihaoyi:upickle-implicits_2.12:0.9.8:default
com.lihaoyi:upickle_2.12:0.9.8:default
io.get-coursier:interface:0.0.8:default
net.java.dev.jna:jna:4.2.2:default
org.javassist:javassist:3.21.0-GA:default
org.jline:jline-reader:3.6.2:default
org.jline:jline-terminal:3.6.2:default
org.jline:jline-terminal-jna:3.6.2:default
org.scala-lang:scala-compiler:2.12.10:default
org.scala-lang:scala-library:2.12.10:default
org.scala-lang:scala-reflect:2.12.10:default
org.scala-lang.modules:scala-collection-compat_2.12:2.1.3:default
org.scala-lang.modules:scala-xml_2.12:1.2.0:default
```
with
```bash
$ cs resolve ammonite --classpath-order=true
com.lihaoyi:ammonite_2.12.10:2.0.4:default
com.lihaoyi:ammonite-terminal_2.12:2.0.4:default
com.lihaoyi:ammonite-ops_2.12:2.0.4:default
com.lihaoyi:ammonite-util_2.12:2.0.4:default
com.lihaoyi:ammonite-runtime_2.12.10:2.0.4:default
com.lihaoyi:ammonite-interp-api_2.12.10:2.0.4:default
com.lihaoyi:ammonite-repl-api_2.12.10:2.0.4:default
com.lihaoyi:ammonite-interp_2.12.10:2.0.4:default
com.lihaoyi:ammonite-repl_2.12.10:2.0.4:default
com.lihaoyi:sourcecode_2.12:0.2.0:default
com.lihaoyi:fansi_2.12:0.2.8:default
com.lihaoyi:os-lib_2.12:0.6.3:default
org.scala-lang.modules:scala-collection-compat_2.12:2.1.3:default
com.lihaoyi:pprint_2.12:0.5.8:default
com.lihaoyi:upickle_2.12:0.9.8:default
com.lihaoyi:requests_2.12:0.5.0:default
org.scala-lang:scala-compiler:2.12.10:default
org.scala-lang:scala-reflect:2.12.10:default
io.get-coursier:interface:0.0.8:default
com.github.scopt:scopt_2.12:3.7.1:default
com.lihaoyi:scalaparse_2.12:2.2.3:default
org.javassist:javassist:3.21.0-GA:default
org.scala-lang.modules:scala-xml_2.12:1.2.0:default
org.jline:jline-terminal:3.6.2:default
org.jline:jline-terminal-jna:3.6.2:default
org.jline:jline-reader:3.6.2:default
com.github.javaparser:javaparser-core:3.2.5:default
com.lihaoyi:geny_2.12:0.5.0:default
org.scala-lang:scala-library:2.12.10:default
com.lihaoyi:ujson_2.12:0.9.8:default
com.lihaoyi:upack_2.12:0.9.8:default
com.lihaoyi:upickle-implicits_2.12:0.9.8:default
com.lihaoyi:fastparse_2.12:2.2.3:default
net.java.dev.jna:jna:4.2.2:default
com.lihaoyi:upickle-core_2.12:0.9.8:default
```

The latter is unreadable past the first few lines IMO.

**Comment by dwijnand** [comment]
Hmm.  I guess at the very minimum `resolve` should express how it's not classpath order, aka the order used by anything else (launch/bootstrap/sbt).  I was thinking maybe even say that on every exec of `resolve`, but maybe that's annoying and the docs is enough.

**Comment by alexarchambault** [comment]
Mentioning it in the doc (help message and / or website) is fine, yes.

**Comment by dwijnand** [comment]
Hmm.  `ResolveOptions` is shared between resolve and fetch.  How do I change the default in a way that:

* uses `true` as the new default for everything except `Resolve`
* keeps all the mechanics in place, specifically parsing, outputting the correct default for each command

?

**Comment by alexarchambault** [comment]
I was thinking about making in option (see https://github.com/coursier/coursier/pull/1546#issuecomment-582871700).

File: modules/cli/src/test/scala/coursier/cli/ResolveTests.scala [hunk 1]
```diff
@@ -342,9 +342,9 @@ class ResolveTests extends FlatSpec with BeforeAndAfterAll {
     val expectedOutput =
       """https://repo1.maven.org/maven2/com/github/alexarchambault/case-app_2.13/2.0.0-M9/case-app_2.13-2.0.0-M9.jar
         |https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.0/scala-library-2.13.0.jar
-        |https://repo1.maven.org/maven2/com/chuusai/shapeless_2.13/2.3.3/shapeless_2.13-2.3.3.jar
         |https://repo1.maven.org/maven2/com/github/alexarchambault/case-app-annotations_2.13/2.0.0-M9/case-app-annotations_2.13-2.0.0-M9.jar
         |https://repo1.maven.org/maven2/com/github/alexarchambault/case-app-util_2.13/2.0.0-M9/case-app-util_2.13-2.0.0-M9.jar
+        |https://repo1.maven.org/maven2/com/chuusai/shapeless_2.13/2.3.3/shapeless_2.13-2.3.3.jar
```
Comment (alexarchambault) [inline modules/cli/src/test/scala/coursier/cli/ResolveTests.scala]: Wondering why this one changes 🤔

Comment (dwijnand) [reply modules/cli/src/test/scala/coursier/cli/ResolveTests.scala]: Yeah, I kept myself from investing even more time diagnosing.  Thanks for having a look.

Comment (alexarchambault) [reply modules/cli/src/test/scala/coursier/cli/ResolveTests.scala]: I had missed that this corresponds to the `--candidate-urls` option. It gets its artifact list on its own, I pushed a commit for it to rely on `--classpath-order` if it's there.

**Comment by alexarchambault** [comment]
Merging, thanks @dwijnand!

**Comment by dwijnand** [comment]
Nice one.  Thank you.

**Pull request merged**